lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1848,16 +1851,19 @@ signTransaction
   -> UTxO
   -- ^ The total UTxO set of the wallet (i.e. if pending transactions all
   -- applied).
+  -> Maybe (Index 'Hardened 'AccountK)
+  -- ^ account ix used only for delegation scripts
   -> SealedTx
   -- ^ The transaction to sign
   -> SealedTx
   -- ^ The original transaction, with additional signatures added where
   -- necessary
-signTransaction tl preferredLatestEra keyLookup (rootKey, rootPwd) utxo =
+signTransaction tl preferredLatestEra witCountCtx keyLookup (rootKey, rootPwd)
+    utxo accIxForStakingM =
     let
         rewardAcnt :: (XPrv, Passphrase "encryption")
         rewardAcnt =
-            (getRawKey $ deriveRewardAccount @k rootPwd rootKey, rootPwd)
+            (getRawKey $ deriveRewardAccount @k rootPwd rootKey minBound, rootPwd)
```

Comment:
I believe `rewardAcnt` is redundant in the case of shared wallets; if we stake the XPrv for signing withdrawals and certificates will always be the same. This doesn't really matter, but could be confusing. I suspect it could be fixed now, but otherwise [ADP-2675](https://input-output.atlassian.net/browse/ADP-2675) should make this easier to deal with.